### PR TITLE
ARROW-14850: [R] Update ARROW_DEPENDENCY_SOURCE to default to AUTO

### PR DESCRIPTION
**Pull request by thisisnic**

### CODE REVIEW

**Comment by thisisnic** [comment]
@github-actions crossbow submit all

**Comment by thisisnic** [comment]
@github-actions crossbow -g r

**Comment by thisisnic** [comment]
@github-actions crossbow submit -g r

File: r/configure [hunk 1]
```diff
@@ -153,8 +153,7 @@ else
       fi
 
       if [ "${ARROW_DEPENDENCY_SOURCE}" = "" ]; then
-        # TODO: BUNDLED is still default for now, but we plan to change it to AUTO
-        ARROW_DEPENDENCY_SOURCE=BUNDLED; export ARROW_DEPENDENCY_SOURCE
+        ARROW_DEPENDENCY_SOURCE=AUTO; export ARROW_DEPENDENCY_SOURCE
```
Comment (nealrichardson) [inline r/configure:156]: (Can't leave this comment below but) on L162, we should revise the message, maybe just

```
echo "**** pkg-config not installed, setting ARROW_DEPENDENCY_SOURCE=BUNDLED"
```

**Review by nealrichardson — COMMENTED** [review]
The other thing we should do, if you haven't already, is document how to install all of the possible dependencies with system packages, which ones can't be installed as system packages, etc. This may be general C++ documentation, but we may also want an R-flavored version since we don't deal with all of the possible C++ dependencies in the R package (some are for features we don't enable, some are test/dev dependencies, etc.). Maybe the docs just start by pointing at `ci/docker` and all of the cpp docker files there: they install dependencies, set ARROW_DEPENDENCY_SOURCE=SYSTEM, and then also set a few other dependency sources as BUNDLED. Some dockerfiles even have comments explaining why (insufficient version, etc.).

File: r/configure (hunk 1 above)
Comment (thisisnic) [reply r/configure:156]: Done!

**Comment by thisisnic** [comment]
> The other thing we should do, if you haven't already, is document how to install all of the possible dependencies with system packages, which ones can't be installed as system packages, etc. This may be general C++ documentation, but we may also want an R-flavored version since we don't deal with all of the possible C++ dependencies in the R package (some are for features we don't enable, some are test/dev dependencies, etc.). Maybe the docs just start by pointing at ci/docker and all of the cpp docker files there: they install dependencies, set ARROW_DEPENDENCY_SOURCE=SYSTEM, and then also set a few other dependency sources as BUNDLED. Some dockerfiles even have comments explaining why (insufficient version, etc.).

OK, will be sure to cover this in https://github.com/apache/arrow/pull/11521

**Review by jonkeane — APPROVED** [review]
This looks good (and will be especially good to check this in CI for a bit before the release).

I've looked through the build logs and both the bundled and system ones seems to be doing what we expect.

**Pull request closed without merging**